Runtime panic because out of touchID

<!-- Please search for open issues that relate to the same problem before opening a new one. -->

### Describe the bug:
<!-- A clear and concise description about the bug. -->
While debugging #2406 , a runtime panic crashes the app. Here is a stack trace:

```
panic: out of touchIDs

goroutine 1 [running, locked to thread]:
github.com/fyne-io/mobile/app.sendTouch(0x130127b98, 0x10404a2a0, 0x30127b98, 0x1)
	/Users/changkun/go/pkg/mod/github.com/fyne-io/mobile@v0.1.3-0.20210811181917-d62fb16c28ec/app/darwin_ios.go:182 +0x184
github.com/fyne-io/mobile/app._Cfunc_runApp()
	_cgo_gotypes.go:201 +0x38
github.com/fyne-io/mobile/app.main(0x130114760)
	/Users/changkun/go/pkg/mod/github.com/fyne-io/mobile@v0.1.3-0.20210811181917-d62fb16c28ec/app/darwin_ios.go:75 +0x58
github.com/fyne-io/mobile/app.Main(...)
	/Users/changkun/go/pkg/mod/github.com/fyne-io/mobile@v0.1.3-0.20210811181917-d62fb16c28ec/app/app.go:21
fyne.io/fyne/v2/internal/driver/gomobile.(*mobileDriver).Run(0x13011b450)
	/Users/changkun/dev/changkun.de/fyne/internal/driver/gomobile/driver.go:138 +0x64
fyne.io/fyne/v2/internal/driver/gomobile.(*window).ShowAndRun(0x104329ae0)
	/Users/changkun/dev/changkun.de/fyne/internal/driver/gomobile/window.go:187 +0x3c
main.main()
	/Users/changkun/dev/changkun.de/fyne/iosrend/main.go:46 +0x2a0
```

It looks like sendTouch may happen concurrently. Adding an RWMutex would be a quick fix.

### To Reproduce:

1. run the reproducer in #2406 
2. keep clicking the button
3. if lucky, the problem occurs

### Device (please complete the following information):
 - **OS:** iOS 15
 - **Version:** beta 6
 - **Go version:** gotip based on CL 344969
 - **Fyne version:** <!-- [e.g. 2.0.3 or git SHA] --> develop 83ca35d98be7cf401579c2f78b53dabd7aea250a

--- a/app/darwin_ios.go
+++ b/app/darwin_ios.go
@@ -154,7 +154,12 @@ func updateConfig(width, height, orientation int32) {
 //
 // It is widely reported that the iPhone can handle up to 5 simultaneous
 // touch events, while the iPad can handle 11.
-var touchIDs [11]uintptr
+var (
+	// touchIDs may arrive concurrently, use a lock for safety.
+	// See https://github.com/fyne-io/fyne/issues/2407.
+	touchMu  sync.RWMutex
+	touchIDs [11]uintptr
+)
 
 var touchEvents struct {
 	sync.Mutex
@@ -164,12 +169,17 @@ var touchEvents struct {
 //export sendTouch
 func sendTouch(cTouch, cTouchType uintptr, x, y float32) {
 	id := -1
+
+	touchMu.RLock()
 	for i, val := range touchIDs {
 		if val == cTouch {
 			id = i
 			break
 		}
 	}
+	touchMu.RUnlock()
+
+	touchMu.Lock()
 	if id == -1 {
 		for i, val := range touchIDs {
 			if val == 0 {
@@ -187,6 +197,7 @@ func sendTouch(cTouch, cTouchType uintptr, x, y float32) {
 	if t == touch.TypeEnd {
 		touchIDs[id] = 0
 	}
+	touchMu.Unlock()
 
 	theApp.eventsIn <- touch.Event{
 		X:        x,

